src/app/mod.rs

```diff
@@ -300,32 +301,75 @@ pub async fn run_app(cli: cli::DftCli, state: state::AppState<'_>) -> Result<()>
     info!("Running app with state: {:?}", state);
     let mut app = App::new(state, cli.clone());
 
-    match &cli.command {
-        Some(cli::Command::App(_)) | None => {
-            app.execute_ddl();
+    app.execute_ddl();
 
-            #[cfg(feature = "flightsql")]
-            app.establish_flightsql_connection();
+    #[cfg(feature = "flightsql")]
+    app.establish_flightsql_connection();
 
-            let mut terminal =
-                ratatui::Terminal::new(CrosstermBackend::new(std::io::stdout())).unwrap();
-            app.enter(true)?;
-            // Main loop for handling events
-            loop {
-                let event = app.next().await?;
+    let mut terminal = ratatui::Terminal::new(CrosstermBackend::new(std::io::stdout())).unwrap();
+    app.enter(true)?;
+    // Main loop for handling events
+    loop {
+        let event = app.next().await?;
 
-                if let AppEvent::Render = event.clone() {
-                    terminal.draw(|f| f.render_widget(&app, f.area()))?;
-                };
+        if let AppEvent::Render = event.clone() {
+            terminal.draw(|f| f.render_widget(&app, f.area()))?;
+        };
 
-                app.handle_app_event(event)?;
+        app.handle_app_event(event)?;
 
-                if app.state.should_quit {
-                    break;
-                }
-            }
-            app.exit()?;
+        if app.state.should_quit {
+            break;
+        }
+    }
+    app.exit()
+}
+
+pub async fn execute_files(files: Vec<PathBuf>, state: &state::AppState<'_>) -> Result<()> {
+    info!("Executing files: {:?}", files);
+    let execution = ExecutionContext::new(state.config.execution.clone());
+
+    for file in files {
+        exec_from_file(&execution, &file).await?
+    }
+
+    Ok(())
+}
+
+/// run and execute SQL statements and commands from a file, against a context
+/// with the given print options
+pub async fn exec_from_file(ctx: &ExecutionContext, file: &Path) -> Result<()> {
```

Comment:
This is the "process commands from a file" code -- it is ased on the code in datafusion-cli